src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java

```diff
@@ -0,0 +1,102 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import com.puppycrawl.tools.checkstyle.internal.utils.XmlUtil;
+import picocli.CommandLine;
+
+public class CliOptionsXdocsSyncTest {
+
+    @Test
+    public void validateCliDocSection() throws Exception {
+        final Map<String, String> cmdDesc = new HashMap<>();
+        final Set<String> ignoreXdocToMain = new HashSet<>();
+        // used to ignore options present in xdocs but not in Main.java
+        ignoreXdocToMain.add("-V");
+        ignoreXdocToMain.add("-h");
+
+        final Path path = Paths.get("src/xdocs/cmdline.xml.vm");
+        final String input = new String(Files.readAllBytes(path), UTF_8);
+
+        final Document document = XmlUtil.getRawXml(path.getFileName().toString(), input, input);
+        final NodeList sections = document.getElementsByTagName("section");
+        for (int i = 0; i < sections.getLength(); i++) {
+            final Node subSection = sections.item(i);
+            final Node name = subSection.getAttributes().getNamedItem("name");
+            if ("Command line usage".equals(name.getNodeValue())) {
+                final List<Node> children =
+                        new ArrayList<>(XmlUtil.getChildrenElements(subSection));
+                for (Node node:children) {
+                    if ("ul".equals(node.getNodeName())) {
+                        final List<Node> listElements =
+                                new ArrayList<>(XmlUtil.getChildrenElements(node));
+                        for (Node element:listElements) {
+                            final String txt = element.getTextContent()
+                                    .trim().replaceAll("\\s+", " ");
+                            cmdDesc.put(txt.substring(0, 2), txt.substring(3));
+                        }
+                    }
+                }
+            }
+        }
+
+        final Class<?> cliOptionsClazz = Class.forName("com.puppycrawl.tools.checkstyle"
+                + ".Main$CliOptions");
+        final Field[] fields = cliOptionsClazz.getDeclaredFields();
+        for (Field field:fields) {
+            final CommandLine.Option optionsAnnotation =
+                  field.getAnnotation(CommandLine.Option.class);
+            if (optionsAnnotation != null) {
+                final String option = optionsAnnotation.names()[0];
+                final String descXdoc = cmdDesc.get(option);
+                assertNotNull(descXdoc, "CLI Option" + option + "present in Main.java not "
+                       + "documented in cmdline.xml.vm");
+                assertEquals(descXdoc, optionsAnnotation.description()[0],
+                        "CLI options descriptions should match");
+                cmdDesc.remove(
+                        optionsAnnotation.names()[0]);
```

Comment:
There is a variable for `optionsAnnotation.names()[0]`

Reply:
Done